chat/chat.c

```diff
@@ -1292,9 +1295,10 @@ int put_string(register char *s)
  *	When called with -1, a '\n' character is generated when
  *	the cursor is not at the beginning of a line.
  */
-void echo_stderr(int n)
+int echo_stderr(int n)
```

Comment:
It looks like that caller of `echo_stderr()` ignores return value, so I'm not sure if it makes sense to change return from void to int if it is always ignored...

Reply:
If the function fail to write to stderr, then all bets are off. The two places this is used is in terminate() and getstring(), the former pretty much terminates the application and the later may have some significance. But either way, if you cannot write to stderr something might be very off and trying to call msgf() with a warning may not do you any good either. 

While good coding practices should include both having a function return a value and also processing this at the next level up, I'd say this is more a part of code evolution. Eventually, functions like chat_send() may end up wanting to return a different value for a different scenario and it'll be one less change to make...

Do you have a better idea? Or should we just not fix this? Perhaps let the user (distribution maintainer) specify -Wno-unused-result and then we can go about our day as usual?

Reply:
Interesting that the code is using both write(2, ... and fprintf(stderr, ...

Clearly we don't want to call fatal() or terminate() from echo_stderr(), since echo_stderr() is called from terminate(). However, arguably the call to echo_stderr() in get_string() could call fatal() if echo_stderr() returns failure.